doxygen/contributor_help_pages/require_meta.md

```diff
@@ -0,0 +1,191 @@
+## Using requires<> for general overloads. {#require_meta_doc}
+
+The [`requires`](@ref require_meta) type traits are aliases for `std::enable_if_t` that have premade
+conditions for turning on and off function definitions during compilation.
+These are useful for having generalized templates while still overloading a function or class.
+You can think of these as "legacy concepts" aka I think the closest thing you can get to C++20 concepts without having the `requires` keyword.
+
+`requires` are [`std::enable_if_t`](https://en.cppreference.com/w/cpp/types/enable_if) aliases such as the below definition of @ref stan::require_t.
+
+```cpp
+template <typename T>
+using require_t = std::enable_if_t<T::value>;
+```
+
+Note that unlike `std::enable_if_t`, the alias @ref stan::require_t 's template type `T`
+must have a valid member `value`.
+
+The most common use cases for a `require` is to overload a function or
+declare specializations of a class. For example, the function below will only work
+on types derived from [`Eigen::DenseBase`](https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html) with only 1 row or column at compile time (aka types such as `Eigen::Matrix<double, -1, 1>` or `Eigen::Matrix<double, 1, -1>`).
+
+```cpp
+template <typename EigVec, require_eigen_vector_t<EigVec>* = nullptr>
+auto my_func(const EigVec& x) {
+  // ...
+}
+```
+
+The `require` is included in the template as a pointer [non-type template parameter](https://en.cppreference.com/w/cpp/language/template_parameters#Non-type_template_parameter) with a default value of `nullptr`. This might look a bit odd, but it uses the fact that any non-type template parameter of type `void*` with a default of `nullptr` will be ignored by the compiler. Under the hood, if any of the `requires` are successful they will return back a type `void`. So when we pass a type that the `require` accepts we get back a `void* = nullptr`, which is safely ignored by the compiler. In the case that the type does not satisfy the `require` then the function is removed from the set of possible functions the caller could use via SFINAE. With this scheme we end up having a very nice pattern for writing generic templates for functions while also being able to restrict the set of types that a function can be used for.
```

Comment:
Didn't you already say this?

Reply:
I did say this in another doc but this is the main doc so I think it would be good to include here